### DISCUSSION TITLE
建议缩短雷神模式的动画长度

### DISCUSSION THREAD

**Discussion by debuggerx01** [linuxdeepin/developer-center#2908] [Features Request & Ideas | 特性请求 & 头脑风暴]
本人强依赖终端的雷神模式，现在雷神模式的显示动画有点过于长了，感觉非常拖沓，建议改为现有动画时长的一半或以下；

相对的，开启窗管的魔灯特效之后系统整体质感有所提升，但是动画时间又太短了，建议改为现有时长的1.5倍左右

（苹果系统下的窗口动画就有点太长了，为了美观牺牲了效率，我就挺不爱用的）

看源码里定义了基础动画时长就有300ms，最长可达450ms，确实太长了：

![image](https://user-images.githubusercontent.com/19624835/144180704-183390cf-d05e-422d-99f0-4e70bbc14e80.png)

参考google MD设计指南推荐的动效时长，桌面环境下的动画一般200ms以内就可以了：

![image](https://user-images.githubusercontent.com/19624835/144180727-a4a8c4c7-20ce-41ae-b444-48f43bb69509.png)

建议分别将基础时间从300改为100，高度分配时间从150改为50：

参考：[https://www.debuggerx.com/2021/01/10/make-deepin-terminal-quake-mode-faster/](https://www.debuggerx.com/2021/01/10/make-deepin-terminal-quake-mode-faster/)

@ch593030323

**Comment by ch593030323** [comment]
Thank u for your suggestion. We will consider this function reasonably.

**Comment by debuggerx01** [comment]
linuxdeepin/deepin-terminal#109

**Comment by debuggerx01** [comment]
any update?

**Reply by BLumia** [reply]
这个 PR 提的较早，是在项目的开发工作流还没迁移到 GitHub 时创建的，我们迁移后可以接受 PR 了，但有一些要求，可以看下 PR 下面的检查，主要是 CLA 以及 commit 信息格式要求（后者我们可以帮着改）。

不过与动画之类相关的修改恐怕要找设计之类的确认过才能决定是否合入，下周我这边帮着确认下看看这种能否合入。如果不能的话，可以考虑加配置项，默认值保持不变，然后允许通过配置的形式控制这个行为。
